crates/runix/src/flake_ref/indirect.rs

```diff
@@ -65,6 +71,17 @@ impl IndirectRef {
             attributes,
         }
     }
+
+    /// Resolves an indirect flake reference to a concrete reference
+    ///
+    /// Note that this method calls `parser-util`, which relies on the `NIX_USER_CONF_FILES`
+    /// environment variable to be set and contain conf files that point to custom registries
+    /// that you want to use for resolution, otherwise only the user's local registry is used.
+    pub fn resolve(&self) -> Result<FlakeRef, UrlParseError> {
+        let json = serde_json::to_string(&self.attributes)?;
+        let resolved = resolve_flake_ref(json, PARSER_UTIL_BIN_PATH)?;
+        FlakeRef::from_parsed(&resolved.resolved_ref)
+    }
```

Comment:
Should this method be tested? Or are you relying on `resolve_flake_ref` and `from_parsed` being tested?

Reply:
The latter. The only new code in this method is converting an indirect ref to JSON, which is what these attributes were parsed from originally. `from_parsed` is now called any time a flake reference is parsed, same re: `resolve_flake_ref`, so those operations are well-tested.